// @typescript-eslint/scope-manager/dist/scope/ScopeType.js
@pattern_meta@
GLIMMER Pattern:
{
  "metadata": {
    "timestamp": "2025-06-16 08:36:04",
    "author": "isdood",
    "pattern_version": "1.0.0",
    "color": "#FF69B4"
  },
  "file_info": {
    "path": "./cursor-extension/node_modules/@typescript-eslint/scope-manager/dist/scope/ScopeType.js",
    "type": "js",
    "hash": "5673975de5b49013dd0d910b104c5b761a7cb22d"
  }
}
@pattern_meta@

"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScopeType = void 0;
var ScopeType;
(function (ScopeType) {
    ScopeType["block"] = "block";
    ScopeType["catch"] = "catch";
    ScopeType["class"] = "class";
    ScopeType["classFieldInitializer"] = "class-field-initializer";
    ScopeType["classStaticBlock"] = "class-static-block";
    ScopeType["conditionalType"] = "conditionalType";
    ScopeType["for"] = "for";
    ScopeType["function"] = "function";
    ScopeType["functionExpressionName"] = "function-expression-name";
    ScopeType["functionType"] = "functionType";
    ScopeType["global"] = "global";
    ScopeType["mappedType"] = "mappedType";
    ScopeType["module"] = "module";
    ScopeType["switch"] = "switch";
    ScopeType["tsEnum"] = "tsEnum";
    ScopeType["tsModule"] = "tsModule";
    ScopeType["type"] = "type";
    ScopeType["with"] = "with";
})(ScopeType || (exports.ScopeType = ScopeType = {}));
