// @typescript-eslint/scope-manager/dist/lib/decorators.js
@pattern_meta@
GLIMMER Pattern:
{
  "metadata": {
    "timestamp": "2025-06-16 08:36:04",
    "author": "isdood",
    "pattern_version": "1.0.0",
    "color": "#FF69B4"
  },
  "file_info": {
    "path": "./cursor-extension/node_modules/@typescript-eslint/scope-manager/dist/lib/decorators.js",
    "type": "js",
    "hash": "140ceda4ba272be16d082b74f8843886a78f6812"
  }
}
@pattern_meta@

"use strict";
// THIS CODE WAS AUTOMATICALLY GENERATED
// DO NOT EDIT THIS CODE BY HAND
// RUN THE FOLLOWING COMMAND FROM THE WORKSPACE ROOT TO REGENERATE:
// npx nx generate-lib @typescript-eslint/scope-manager
Object.defineProperty(exports, "__esModule", { value: true });
exports.decorators = void 0;
const base_config_1 = require("./base-config");
exports.decorators = {
    ClassMemberDecoratorContext: base_config_1.TYPE,
    DecoratorContext: base_config_1.TYPE,
    ClassDecoratorContext: base_config_1.TYPE,
    ClassMethodDecoratorContext: base_config_1.TYPE,
    ClassGetterDecoratorContext: base_config_1.TYPE,
    ClassSetterDecoratorContext: base_config_1.TYPE,
    ClassAccessorDecoratorContext: base_config_1.TYPE,
    ClassAccessorDecoratorTarget: base_config_1.TYPE,
    ClassAccessorDecoratorResult: base_config_1.TYPE,
    ClassFieldDecoratorContext: base_config_1.TYPE,
};
