// @typescript-eslint/scope-manager/dist/lib/es2015.collection.js
@pattern_meta@
GLIMMER Pattern:
{
  "metadata": {
    "timestamp": "2025-06-16 08:36:04",
    "author": "isdood",
    "pattern_version": "1.0.0",
    "color": "#FF69B4"
  },
  "file_info": {
    "path": "./cursor-extension/node_modules/@typescript-eslint/scope-manager/dist/lib/es2015.collection.js",
    "type": "js",
    "hash": "c33561bda50ad7a61ef12495838b8216d92d95e6"
  }
}
@pattern_meta@

"use strict";
// THIS CODE WAS AUTOMATICALLY GENERATED
// DO NOT EDIT THIS CODE BY HAND
// RUN THE FOLLOWING COMMAND FROM THE WORKSPACE ROOT TO REGENERATE:
// npx nx generate-lib @typescript-eslint/scope-manager
Object.defineProperty(exports, "__esModule", { value: true });
exports.es2015_collection = void 0;
const base_config_1 = require("./base-config");
exports.es2015_collection = {
    Map: base_config_1.TYPE_VALUE,
    MapConstructor: base_config_1.TYPE,
    ReadonlyMap: base_config_1.TYPE,
    WeakMap: base_config_1.TYPE_VALUE,
    WeakMapConstructor: base_config_1.TYPE,
    Set: base_config_1.TYPE_VALUE,
    SetConstructor: base_config_1.TYPE,
    ReadonlySet: base_config_1.TYPE,
    WeakSet: base_config_1.TYPE_VALUE,
    WeakSetConstructor: base_config_1.TYPE,
};
