@pattern_meta@
GLIMMER Pattern:
{
  "metadata": {
    "timestamp": "2025-06-16 08:36:04",
    "author": "isdood",
    "pattern_version": "1.0.0",
    "color": "#FF69B4"
  },
  "file_info": {
    "path": "./cursor-extension/node_modules/@typescript-eslint/eslint-plugin/docs/rules/consistent-type-definitions.md",
    "type": "md",
    "hash": "d2daf1af2cb900a972d13bab1b61d56b118619f2"
  }
}
@pattern_meta@

---
description: 'Enforce type definitions to consistently use either `interface` or `type`.'
---

> üõë This file is source code, not the primary documentation location! üõë
>
> See **https://typescript-eslint.io/rules/consistent-type-definitions** for documentation.

TypeScript provides two common ways to define an object type: `interface` and `type`.

```ts
// type alias
type T1 = {
  a: string;
  b: number;
};

// interface keyword
interface T2 {
  a: string;
  b: number;
}
```

The two are generally very similar, and can often be used interchangeably.
Using the same type declaration style consistently helps with code readability.

## Options

- `"interface"` _(default)_: enforce using `interface`s for object type definitions.
- `"type"`: enforce using `type`s for object type definitions.

### `interface`

<!--tabs-->

#### ‚ùå Incorrect

```ts
/* eslint @typescript-eslint/consistent-type-definitions: ["error", "interface"] */

type T = { x: number };
```

#### ‚úÖ Correct

```ts
/* eslint @typescript-eslint/consistent-type-definitions: ["error", "interface"] */

type T = string;
type Foo = string | {};

interface T {
  x: number;
}
```

### `type`

<!--tabs-->

#### ‚ùå Incorrect

```ts
/* eslint @typescript-eslint/consistent-type-definitions: ["error", "type"] */

interface T {
  x: number;
}
```

#### ‚úÖ Correct

```ts
/* eslint @typescript-eslint/consistent-type-definitions: ["error", "type"] */

type T = { x: number };
```

## When Not To Use It

If you specifically want to use an interface or type literal for stylistic reasons, you can disable this rule.
