@pattern_meta@
GLIMMER Pattern:
{
  "metadata": {
    "timestamp": "2025-06-16 08:36:04",
    "author": "isdood",
    "pattern_version": "1.0.0",
    "color": "#FF69B4"
  },
  "file_info": {
    "path": "./cursor-extension/node_modules/@typescript-eslint/eslint-plugin/docs/rules/consistent-indexed-object-style.md",
    "type": "md",
    "hash": "ea51c21648713f578e10863cc37a0de5896e950c"
  }
}
@pattern_meta@

---
description: 'Require or disallow the `Record` type.'
---

> üõë This file is source code, not the primary documentation location! üõë
>
> See **https://typescript-eslint.io/rules/consistent-indexed-object-style** for documentation.

TypeScript supports defining arbitrary object keys using an index signature. TypeScript also has a builtin type named `Record` to create an empty object defining only an index signature. For example, the following types are equal:

```ts
interface Foo {
  [key: string]: unknown;
}

type Foo = {
  [key: string]: unknown;
};

type Foo = Record<string, unknown>;
```

Keeping to one declaration form consistently improve code readability.

## Options

- `"record"` _(default)_: only allow the `Record` type.
- `"index-signature"`: only allow index signatures.

### `record`

<!--tabs-->

#### ‚ùå Incorrect

```ts
/* eslint @typescript-eslint/consistent-indexed-object-style: ["error", "record"] */

interface Foo {
  [key: string]: unknown;
}

type Foo = {
  [key: string]: unknown;
};
```

#### ‚úÖ Correct

```ts
/* eslint @typescript-eslint/consistent-indexed-object-style: ["error", "record"] */

type Foo = Record<string, unknown>;
```

### `index-signature`

<!--tabs-->

#### ‚ùå Incorrect

```ts
/* eslint @typescript-eslint/consistent-indexed-object-style: ["error", "index-signature"] */

type Foo = Record<string, unknown>;
```

#### ‚úÖ Correct

```ts
/* eslint @typescript-eslint/consistent-indexed-object-style: ["error", "index-signature"] */

interface Foo {
  [key: string]: unknown;
}

type Foo = {
  [key: string]: unknown;
};
```
