@pattern_meta@
GLIMMER Pattern:
{
  "metadata": {
    "timestamp": "2025-06-16 08:36:04",
    "author": "isdood",
    "pattern_version": "1.0.0",
    "color": "#FF69B4"
  },
  "file_info": {
    "path": "./cursor-extension/node_modules/@typescript-eslint/eslint-plugin/docs/rules/prefer-regexp-exec.md",
    "type": "md",
    "hash": "dc7af3a5347105a79a303950bd213b436e510048"
  }
}
@pattern_meta@

---
description: 'Enforce `RegExp#exec` over `String#match` if no global flag is provided.'
---

> üõë This file is source code, not the primary documentation location! üõë
>
> See **https://typescript-eslint.io/rules/prefer-regexp-exec** for documentation.

`String#match` is defined to work the same as `RegExp#exec` when the regular expression does not include the `g` flag.
Keeping to consistently using one of the two can help improve code readability.

This rule reports when a `String#match` call can be replaced with an equivalent `RegExp#exec`.

> `RegExp#exec` may also be slightly faster than `String#match`; this is the reason to choose it as the preferred usage.

## Examples

<!--tabs-->

### ‚ùå Incorrect

```ts
'something'.match(/thing/);

'some things are just things'.match(/thing/);

const text = 'something';
const search = /thing/;
text.match(search);
```

### ‚úÖ Correct

```ts
/thing/.exec('something');

'some things are just things'.match(/thing/g);

const text = 'something';
const search = /thing/;
search.exec(text);
```

## When Not To Use It

If you prefer consistent use of `String#match` for both with `g` flag and without it, you can turn this rule off.
