@pattern_meta@
GLIMMER Pattern:
{
  "metadata": {
    "timestamp": "2025-06-16 08:36:04",
    "author": "isdood",
    "pattern_version": "1.0.0",
    "color": "#FF69B4"
  },
  "file_info": {
    "path": "./cursor-extension/node_modules/@typescript-eslint/eslint-plugin/docs/rules/TEMPLATE.md",
    "type": "md",
    "hash": "bed002d1821ed4003f17e1c8528c3889a1bb27e2"
  }
}
@pattern_meta@

> üõë This file is source code, not the primary documentation location! üõë
>
> See **https://typescript-eslint.io/rules/your-rule-name** for documentation.

## Examples

To fill out: tell us more about this rule.

<!--tabs-->

### ‚ùå Incorrect

```ts
// To fill out: incorrect code
```

### ‚úÖ Correct

```ts
// To fill out: correct code
```

## When Not To Use It

To fill out: why wouldn't you want to use this rule?
For example if this rule requires a feature released in a certain TS version.
