// @typescript-eslint/eslint-plugin/dist/index.js
@pattern_meta@
GLIMMER Pattern:
{
  "metadata": {
    "timestamp": "2025-06-16 08:36:04",
    "author": "isdood",
    "pattern_version": "1.0.0",
    "color": "#FF69B4"
  },
  "file_info": {
    "path": "./cursor-extension/node_modules/@typescript-eslint/eslint-plugin/dist/index.js",
    "type": "js",
    "hash": "158aad14fea842b7a486936d412b46ddaf121ad0"
  }
}
@pattern_meta@

"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
const all_1 = __importDefault(require("./configs/all"));
const base_1 = __importDefault(require("./configs/base"));
const eslint_recommended_1 = __importDefault(require("./configs/eslint-recommended"));
const recommended_1 = __importDefault(require("./configs/recommended"));
const recommended_requiring_type_checking_1 = __importDefault(require("./configs/recommended-requiring-type-checking"));
const strict_1 = __importDefault(require("./configs/strict"));
const rules_1 = __importDefault(require("./rules"));
module.exports = {
    rules: rules_1.default,
    configs: {
        all: all_1.default,
        base: base_1.default,
        recommended: recommended_1.default,
        'eslint-recommended': eslint_recommended_1.default,
        'recommended-requiring-type-checking': recommended_requiring_type_checking_1.default,
        strict: strict_1.default,
    },
};
