// @typescript-eslint/scope-manager/dist/definition/DefinitionType.js
@pattern_meta@
GLIMMER Pattern:
{
  "metadata": {
    "timestamp": "2025-06-16 08:36:04",
    "author": "isdood",
    "pattern_version": "1.0.0",
    "color": "#FF69B4"
  },
  "file_info": {
    "path": "./cursor-extension/node_modules/@typescript-eslint/scope-manager/dist/definition/DefinitionType.js",
    "type": "js",
    "hash": "311d8d514f5eb55ca651acc979ed79d24408b89d"
  }
}
@pattern_meta@

"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DefinitionType = void 0;
var DefinitionType;
(function (DefinitionType) {
    DefinitionType["CatchClause"] = "CatchClause";
    DefinitionType["ClassName"] = "ClassName";
    DefinitionType["FunctionName"] = "FunctionName";
    DefinitionType["ImplicitGlobalVariable"] = "ImplicitGlobalVariable";
    DefinitionType["ImportBinding"] = "ImportBinding";
    DefinitionType["Parameter"] = "Parameter";
    DefinitionType["TSEnumName"] = "TSEnumName";
    DefinitionType["TSEnumMember"] = "TSEnumMemberName";
    DefinitionType["TSModuleName"] = "TSModuleName";
    DefinitionType["Type"] = "Type";
    DefinitionType["Variable"] = "Variable";
})(DefinitionType || (exports.DefinitionType = DefinitionType = {}));
